src/languages/typescript.js

```diff
@@ -28,10 +28,18 @@ export default function(hljs) {
     "unknown"
   ];
   const NAMESPACE = {
-    beginKeywords: 'namespace',
+    begin: [
+      /namespace/,
+      /\s+/,
+      hljs.IDENT_RE
+    ],
+    beginScope: {
+      1: "keyword",
+      3: "variable"
+    },
     end: /\{/,
     excludeEnd: true,
-    contains: [ tsLanguage.exports.CLASS_REFERENCE ]
+    contains: [tsLanguage.exports.CLASS_REFERENCE]
```

Comment:
Not sure we need any of this anymore - the match rule should be enough to highlight the keyword and namespace name... but should IDENT_RE used above to recognize namespace perhaps be `CLASS_REFERENCE[:match]` instead - if the same names rules apply.

Reply:
```
const NAMESPACE = {
    begin: [
      /namespace/,
      /\s+/,
      hljs.IDENT_RE
    ],
    beginScope: {
      1: "keyword",
      3: "title.class"
    }
  };
```

Is the above change enough or i should change something else.